sdk/python/feast/repo_operations.py

```diff
@@ -145,134 +141,98 @@ def apply_total(repo_config: RepoConfig, repo_path: Path, skip_source_validation
         )
+    # TODO: delete entities from the registry too
+    registry.commit()
 
-    # Create views that should exist
-    for view in repo.feature_views:
-        registry.apply_feature_view(view, project, commit=False)
+    # Add / update views + entities
+    store.apply(entities_to_keep + views_to_keep)
+    for entity in entities_to_keep:
+        click.echo(
+            f"Registered entity {Style.BRIGHT + Fore.GREEN}{entity.name}{Style.RESET_ALL}"
+        )
+    for view in views_to_keep:
         click.echo(
             f"Registered feature view {Style.BRIGHT + Fore.GREEN}{view.name}{Style.RESET_ALL}"
         )
-    registry.commit()
 
     apply_feature_services(registry, project, repo)
 
     infra_provider = get_provider(repo_config, repo_path)
-
-    all_to_delete: List[Union[FeatureTable, FeatureView]] = []
-    all_to_delete.extend(tables_to_delete)
-    all_to_delete.extend(views_to_delete)
-
-    all_to_keep: List[Union[FeatureTable, FeatureView]] = []
-    all_to_keep.extend(repo.feature_tables)
-    all_to_keep.extend(repo.feature_views)
-
-    entities_to_delete: List[Entity] = []
-    repo_entities_names = set([e.name for e in repo.entities])
-    for registry_entity in registry.list_entities(project=project):
-        if registry_entity.name not in repo_entities_names:
-            entities_to_delete.append(registry_entity)
-
-    entities_to_keep: List[Entity] = repo.entities
-
-    for name in [view.name for view in repo.feature_tables] + [
-        table.name for table in repo.feature_views
-    ]:
+    for name in [view.name for view in views_to_keep]:
         click.echo(
             f"Deploying infrastructure for {Style.BRIGHT + Fore.GREEN}{name}{Style.RESET_ALL}"
         )
-    for name in [view.name for view in views_to_delete] + [
-        table.name for table in tables_to_delete
-    ]:
+    for name in [view.name for view in views_to_delete]:
         click.echo(
             f"Removing infrastructure for {Style.BRIGHT + Fore.GREEN}{name}{Style.RESET_ALL}"
         )
+    # TODO: consider echoing also entities being deployed/removed
 
     infra_provider.update_infra(
         project,
-        tables_to_delete=all_to_delete,
-        tables_to_keep=all_to_keep,
+        tables_to_delete=views_to_delete,
+        tables_to_keep=views_to_keep,
         entities_to_delete=entities_to_delete,
         entities_to_keep=entities_to_keep,
         partial=False,
     )
 
 
+def _tag_registry_entities_for_keep_delete(project, registry, repo):
+    entities_to_keep: List[Entity] = repo.entities
+    entities_to_delete: List[Entity] = []
+    repo_entities_names = set([e.name for e in repo.entities])
+    for registry_entity in registry.list_entities(project=project):
+        if registry_entity.name not in repo_entities_names:
+            entities_to_delete.append(registry_entity)
+    return entities_to_delete, entities_to_keep
```

Comment:
Nit: the function is named `_for_keep_delete`, but the lists returned are for deleting and keeping, respectively (i.e. the order is mixed up).